setup.py

```diff
@@ -1,73 +1,18 @@
-import io
-
-from setuptools import find_packages
+# Stub setup.py.
+#
+# XXX: This is deprecated in favor of PEP-517-aware tools and should probably be deleted.
+#
+# The canonical way to build an sdist and wheel is now:
+#
+#     pip install build
+#     python -m build
+#
+# NB: Recent versions of pip are PEP-517-aware so that:
```

Comment:
Exactly so it could be reasonable to just delete it and assume that the contributors are capable of upgrading their pip.

Reply:
If there is consensus on that, I'm more than happy to delete the setup.py.

Reply:
Please no, I still want to be able to do editable installs.

Reply:
I don't completely understand how it works, but recent versions of `pip` will perform editable installs just fine without a `setup.py`.

Reply:
If a setup.cfg but no setup.py exists then pip will basically just create a stub like this setup.py here

Reply:
Leaving the removal of `setup.py` for a subsequent PR.